dask/dataframe/utils.py

```diff
@@ -952,3 +952,14 @@ def valid_divisions(divisions):
         return False
 
     return True
+
+
+def drop_by_shallow_copy(df, columns):
+    """ Use shallow copy to drop columns in place
+    """
+    df2 = df.copy(deep=False)
```

Comment:
This will modify `df`. Is that desired / expected?

Reply:
Can you say more about how this will modify `df`?

Reply:
When we drop the column from `df2`, the column is not dropped from `df`...

```python
In [1]: import pandas as pd; import numpy as np                                                                                                                                                           

In [2]: df = pd.DataFrame(np.random.randn(10, 3), columns=list("ABC"))                                                                                                                                                           

In [3]: df2 = df.copy(deep=False)                                                                                                                                                                                                

In [4]: del df2["A"]                                                                                                                                                                             

In [5]: df2                                                                                                                                                                                                                     
Out[5]: 
          B         C
0 -0.696879  0.263415
1  0.437084  0.193037
2  0.922138 -1.421429
3  1.989285 -0.034092
4  0.361518 -0.622257
5 -0.804664  0.475421
6  1.523969  1.151729
7 -0.090899  0.711572
8  0.126784 -0.003118
9 -0.533765  0.191923

In [6]: df                                                                                                                                                                                                                      
Out[6]: 
          A         B         C
0 -1.243505 -0.696879  0.263415
1  1.125122  0.437084  0.193037
2 -0.807596  0.922138 -1.421429
3 -0.375252  1.989285 -0.034092
4 -0.083836  0.361518 -0.622257
5 -1.400787 -0.804664  0.475421
6 -1.970369  1.523969  1.151729
7  0.063429 -0.090899  0.711572
8 -1.778425  0.126784 -0.003118
9 -0.013437 -0.533765  0.191923
```

Are you refering to the fact that df2 will now be referencing the same data as df after the shallow copy?

Reply:
Whoops, I messed up when looking at this locally. Deleted from the wrong object. You can ignore this.